app/models/tweet.rb

```diff
@@ -0,0 +1,9 @@
+class Tweet < ApplicationRecord
+  belongs_to :user, foreign_key: 'id_user'
+  has_many :bookmarks
+  has_many :likes
+  has_many :hashtags
+  belongs_to :retweeted_tweet, class_name: 'Tweet', foreign_key: 'retweet', optional: true
```

Comment:
the retweet column is boolean and it should be integer so it can be used as foreign key